File: code/leetcode/3.LongestSubstringWithoutCharacters/LongestSubstringWithoutRepeatCharacter.js
var lengthOfLongestSubstring = function(s) {
    var len = s.length;
    var right=0,left=0;
    var maxLen = 0;
    var map = new  Map();
    for (let right = 0; right < s.length; right++) {
        var ch = s[right];
        if(map.has(ch))
            left = Math.max(left,map.get(s[right])+1);
        map.set(ch,right);
        maxLen = Math.max(maxLen,right-left+1);
        
    }
    return maxLen;
};

console.info(lengthOfLongestSubstring("bbbbb"));

File: code/leetcode/3.LongestSubstringWithoutCharacters/LongestSubstringWithoutRepeatCharacter.java
import java.util.HashMap;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int len = s.length();
        int right=0,left=0,maxLen=0;
        var map = new HashMap<Character,Integer>();
        for(right=0;right<len;right++){
            char ch= s.charAt(right);
            if(map.containsKey(ch))
                left = Math.max(map.get(ch)+1,left);

            map.put(ch,right);
            maxLen = Math.max(maxLen,right-left+1);
        }
        return maxLen;
    }
}